opa-manager-operator/src/manifests.py

```diff
@@ -0,0 +1,141 @@
+from typing import Dict
+
+from lightkube.codecs import from_dict
+from ops.manifests import ManifestLabel, Manifests, Patch, SubtractEq
+
+log = logging.getLogger(__file__)
+
+audit_controller = from_dict(
+    dict(
+        apiVersion="apps/v1",
+        kind="Deployment",
+        metadata=dict(name="gatekeeper-audit", namespace="gatekeeper-system"),
+    )
+)
+
+controller_manager = from_dict(
+    dict(
+        apiVersion="apps/v1",
+        kind="Deployment",
+        metadata=dict(
+            name="gatekeeper-controller-manager", namespace="gatekeeper-system"
+        ),
+    )
+)
+
+gatekeeper_system_ns = from_dict(
+    dict(
+        apiVersion="v1",
+        kind="Namespace",
+        metadata=dict(name="gatekeeper-system"),
+    )
+)
+
+
+class ModelNamespace(Patch):
+    """Update the namespace of any namespaced resources to the model name."""
+
+    def __call__(self, obj):
+        if not obj.metadata.namespace:
+            return
+
+        log.info(f"Patching namespace for {obj.kind} {obj.metadata.name}")
+        obj.metadata.namespace = self.manifests.model.name
+
+
+class WebhookConfiguration(Patch):
+    """Update the namespace of any webhook clientConfig services to the model name."""
+
+    def __call__(self, obj):
+        if (
+            obj.kind == "ValidatingWebhookConfiguration"
+            or obj.kind == "MutatingWebhookConfiguration"
+        ):
+            for webhook in obj.webhooks:
+                log.info(
+                    f"Patching clientConfig service namespace for {obj.kind} {obj.metadata.name}"
+                )
+                webhook.clientConfig.service.namespace = self.manifests.model.name
+
+
+class RoleBinding(Patch):
+    """Update the namespace of any RoleBinding or ClusteRoleBinding subjects to the model name."""
+
+    def __call__(self, obj):
+        if obj.kind == "RoleBinding" or obj.kind == "ClusterRoleBinding":
+            for subject in obj.subjects:
+                log.info(
+                    f"Patching subject namespace for {obj.kind} {obj.metadata.name}"
+                )
+                subject.namespace = self.manifests.model.name
+
+
+class ServicePorts(Patch):
+    """Patch target ports to use port numbers instead of named ports.
+    This is an alternative to patching the container ports in the juju-deployed
+    statefulset for the charm
+    """
+
+    def __call__(self, obj):
+        if obj.metadata.name == "gatekeeper-webhook-service":
+            for port in obj.spec.ports:
+                if port.name == "https-webhook-server":
+                    log.info(f"Patching target port for {obj.metadata.name}")
+                    port.targetPort = 8443
+
+
+class ServiceSelector(Patch):
+    """Patch the service selector to match the pod's labels"""
+
+    def __call__(self, obj):
+        if obj.metadata.name == "gatekeeper-webhook-service":
+            obj.spec.selector = {
+                "app.kubernetes.io/name": "gatekeeper-controller-manager"
```

Comment:
It would be nice if we could get the app name dynamically here:
```suggestion
                "app.kubernetes.io/name": self.manifests.model.app.name
```